lite/errors/errors.go

```diff
@@ -3,90 +3,110 @@ package errors
+func (e errUnexpectedValidators) Error() string {
+	return fmt.Sprintf("Validator set is different. Got %X want %X",
+		e.got, e.want)
 }
 
-// ErrValidatorsChanged indicates that the validator set was changed between two commits.
-func ErrValidatorsChanged() error {
-	return errors.WithStack(errValidatorsChanged)
+type errTooMuchChange struct{}
+
+func (e errTooMuchChange) Error() string {
+	return "Insufficient signatures to validate due to valset changes"
 }
 
-// IsTooMuchChangeErr checks whether an error is due to too much change
-// between these validators sets.
-func IsTooMuchChangeErr(err error) bool {
-	return err != nil && (errors.Cause(err) == errTooMuchChange)
+type errMissingValidators struct {
+	chainID string
+	height  int64
 }
 
-// ErrTooMuchChange indicates that the underlying validator set was changed by >1/3.
-func ErrTooMuchChange() error {
-	return errors.WithStack(errTooMuchChange)
+func (e errMissingValidators) Error() string {
+	return fmt.Sprintf("Validators are unknown or missing for chain %s and height %d",
+		e.chainID, e.height)
 }
 
-// IsPastTimeErr ...
-func IsPastTimeErr(err error) bool {
-	return err != nil && (errors.Cause(err) == errPastTime)
+//----------------------------------------
+// Methods for above error types
+
+//-----------------
+// ErrCommitNotFound
+
+// ErrCommitNotFound indicates that a the requested commit was not found.
+func ErrCommitNotFound() error {
+	return cmn.ErrorWrap(errCommitNotFound{}, "")
 }
 
-// ErrPastTime ...
-func ErrPastTime() error {
-	return errors.WithStack(errPastTime)
+func IsErrCommitNotFound(err error) bool {
+	if err_, ok := err.(cmn.Error); ok {
+		_, ok := err_.Data().(errCommitNotFound)
+		return ok
+	}
+	return false
 }
 
-// IsNoPathFoundErr checks whether an error is due to no path of
-// validators in provider from where we are to where we want to be
-func IsNoPathFoundErr(err error) bool {
-	return err != nil && (errors.Cause(err) == errNoPathFound)
+//-----------------
+// ErrUnexpectedValidators
+
+// ErrUnexpectedValidators indicates a validator set mismatch.
+func ErrUnexpectedValidators(got, want []byte) error {
+	return cmn.ErrorWrap(errUnexpectedValidators{
+		got:  got,
+		want: want,
+	}, "")
 }
 
-// ErrNoPathFound ...
-func ErrNoPathFound() error {
-	return errors.WithStack(errNoPathFound)
+func IsErrUnexpectedValidators(err error) bool {
+	if err_, ok := err.(cmn.Error); ok {
+		_, ok := err_.Data().(errUnexpectedValidators)
+		return ok
+	}
+	return false
 }
 
-//--------------------------------------------
+//-----------------
+// ErrTooMuchChange
 
-type errHeightMismatch struct {
-	h1, h2 int64
+// ErrTooMuchChange indicates that the underlying validator set was changed by >1/3.
+func ErrTooMuchChange() error {
+	return cmn.ErrorWrap(errTooMuchChange{}, "")
 }
 
-func (e errHeightMismatch) Error() string {
-	return fmt.Sprintf("Blocks don't match - %d vs %d", e.h1, e.h2)
+func IsErrTooMuchChange(err error) bool {
```

Comment:
Remove.

Reply:
Leaving as is for now, as it *should* be used, and there's a TODO issue to be made and assigned to me.  Will look at this after merge.